src/codecs/processor.ts

```diff
@@ -0,0 +1,194 @@
+import { proxy } from 'comlink';
```

Comment:
This is the main-thread part of processor-worker. It exposes all the image processing methods, and cancels any pending jobs.